src/include/serial.i

```diff
@@ -154,7 +154,7 @@ __col_append_serial_func(WT_SESSION_IMPL *session, WT_INSERT_HEAD *ins_head,
 static inline int
 __wt_col_append_serial(WT_SESSION_IMPL *session, WT_PAGE *page,
     WT_INSERT_HEAD *ins_head, WT_INSERT ***ins_stack, WT_INSERT **new_insp,
-    size_t new_ins_size, uint64_t *recnop, u_int skipdepth)
+    size_t new_ins_size, uint64_t *recnop, u_int skipdepth, bool exclusive)
```

Comment:
Note that these `exclusive` changes are required to avoid self-deadlock where the update/restore path is holding a page lock while re-applying changes.  Now they share the same locks, we need to skip page locking in that path.

I don't love it, very open to better ideas.